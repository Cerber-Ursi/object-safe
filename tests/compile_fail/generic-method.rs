mod test {

use object_safe::object_safe;

#[object_safe]
pub trait Test {
    fn test<T>(&self, _: T);
    fn non_generic(&self);
}

pub struct TestStruct;
impl Test for TestStruct {
    fn test<T>(&self, _: T) {}
    fn non_generic(&self) {}
}

}

use test::{TestStruct, ObjectSafeTest};
fn main() {
    let test = TestStruct;
    test.test();
    test.non_generic();
}