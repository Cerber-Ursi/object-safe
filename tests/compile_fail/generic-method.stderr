error[E0599]: no method named `test` found for type `test::TestStruct` in the current scope
  --> $DIR/generic-method.rs:22:10
   |
11 | pub struct TestStruct;
   | ---------------------- method `test` not found for this
...
22 |     test.test();
   |          ^^^^
   |
   = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope, perhaps add a `use` for it:
   |
19 | use crate::test::Test;
   |

For more information about this error, try `rustc --explain E0599`.
